src/Symfony/Component/Workflow/Event/GuardEvent.php

```diff
@@ -11,21 +11,52 @@
 
 namespace Symfony\Component\Workflow\Event;
 
+use Symfony\Component\Workflow\Marking;
+use Symfony\Component\Workflow\Transition;
+use Symfony\Component\Workflow\TransitionBlocker;
+use Symfony\Component\Workflow\TransitionBlockerList;
+
 /**
  * @author Fabien Potencier <fabien@symfony.com>
  * @author Grégoire Pineau <lyrixx@lyrixx.info>
  */
 class GuardEvent extends Event
 {
-    private $blocked = false;
+    private $transitionBlockerList;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function __construct($subject, Marking $marking, Transition $transition, $workflowName = 'unnamed')
+    {
+        parent::__construct($subject, $marking, $transition, $workflowName);
+
+        $this->transitionBlockerList = new TransitionBlockerList();
+    }
+
+    public function isBlocked(): bool
+    {
+        return !$this->transitionBlockerList->isEmpty();
+    }
+
+    public function setBlocked(bool $blocked): void
+    {
+        if (!$blocked) {
+            $this->transitionBlockerList->reset();
+
+            return;
+        }
+
+        $this->transitionBlockerList->add(TransitionBlocker::createUnknown());
```

Comment:
I think we should not add an unknown reason, if the list is not empty

Reply:
But I will lead to something wrong. 
I disagree with that.

Reply:
For an end user having an unknown reason presented alongside „normal“ reasons is kinda of no value. And retrying automated workflows cant fix unknown reasons.

If we add unknown reason anyway we should at least prevent duplicated unknown reasons.

Reply:
Also I agree that if we were to remove unknown reasons, if their are other reasons, we must do it after all guards are called